modules/log.cpp

```diff
@@ -457,12 +457,26 @@ CModule::EModRet CLogMod::OnSendToIRCMessage(CMessage& Message) {
     return CONTINUE;
 }
 
-void CLogMod::OnJoin(const CNick& Nick, CChan& Channel) {
-    if (NeedJoins()) {
-        PutLog("*** Joins: " + Nick.GetNick() + " (" + Nick.GetIdent() + "@" +
-                   Nick.GetHost() + ")",
-               Channel);
+void CLogMod::OnJoinMessage(CJoinMessage& Message) {
+    if (!NeedJoins())
+        return;
+
+    const CNick& Nick = Message.GetNick();
+    CChan& Channel = *Message.GetChan();
+    CString Account = Message.GetTag("account");
+    const char* s = " ";
+
+    if (Account.empty())
+        Account = Message.GetParam(1);
+
+    if (Account.empty() || Account == "*") {
+        Account = "";
+        s = "";
     }
+
+    PutLog("*** Joins: " + Nick.GetNick() + " (" + Nick.GetIdent() + "@" +
+           Nick.GetHost() + ")" + s + Account,
```

Comment:
Not very obvious, but at least it's there.  I think an overriding concern is to avoid impacting on any existing attempt at parsing the log line as much as possible, so putting it at the end makes sense in that respect.